### ISSUE TITLE
[Bug] Android Review Rejected Unsafe Cipher Mode

### ISSUE THREAD

**Issue by rafetkhallaf** [ammarahm-ed/react-native-mmkv-storage#194]
Hello all,

I'm not sure if this but is 100% related to this library, but it's the only place where i found the code that is causing the issue. So if it's not, I apologize in advance.

I've used this library before on another application a while back and it was accepted and listed on the Android store. Since then I believe Android updated it's security requirements.

Apparently Android doesn't accept ``` getInstance("AES"); ``` and the only place I was able to find this in my code is in this library within SecureKeystore.java file.

The screenshot attached is from the Android Play Console, and the link mentioned on the screenshot is https://support.google.com/faqs/answer/10046138

<img width="706" alt="Screenshot 2021-12-13 102112" src="https://user-images.githubusercontent.com/26126456/145867093-1892bb6c-a186-431a-9653-3c0102eb7292.png">

I'm using react 0.66 and the library version installed is 0.6.6

- Has anybody seen this lately?
- Could changing this to ``` getInstance(Constants.AES_ALGORITHM); ``` fix the problem?

If any additional information needed please let me know. Till then I will be testing changing the code as mentioned on the second point and will update this if it works.

Regards,

**Comment by bolyesz** [comment]
Hi, it seems that we are having the same problem in our app. We were not rejected because of this, but we have the same error in the Pre-launch report details. The first time this error was reported was when we implemented this library.

**Comment by rafetkhallaf** [comment]
I've tested replacing the code as I mentioned in my report, but it still didn't work. We still got the same error message.

@bolyesz , Was your app accepted by Google?

Note: We haven't promoted our app pass internal testing yet, it is still in QA, but just wondering if this is going to prevent the app release on the Google Play Store.

**Comment by kylegillen** [comment]
I also just released the exact same error, and the app was rejected by Google.

Possible to get a fix?

**Comment by rafetkhallaf** [comment]
I've just uploaded a new version to Internal Testing with a small change, let's see if it fixes the issue or not. I'll report the updates here.

**Comment by ammarahm-ed** [comment]
@rafetkhallaf The issue is happening because of [secure-preferences](https://github.com/scottyab/secure-preferences) library which uses less secure encryption algorithm. We will need to remove it. However this means that we will have to target API 23 minimum.

**Comment by rafetkhallaf** [comment]
@ammarahm-ed I see. Well, this is now a critical requirement from Google, so I think it's a something the we, the community, must do, at least releasing a new version.

If Google is going to reject the app the adaptation of this library would be impractical. And this library is actually great and essential to most work flows.

Anything I can help with, I'll be happy to, unfortunately I'm not an Android expert, so It has to be one of our great contributors.

**Comment by bolyesz** [comment]
> @bolyesz , Was your app accepted by Google?

Yes, we were able to release it, in fact it has been like that for the last 4 version of our app, the error was present in the pre-launch report

**Comment by bolyesz** [comment]
> However this means that we will have to target API 23 minimum.

Is this a big problem though? API 23 was initially released in 2015, is it justifiable to support anything older than that?

**Comment by ammarahm-ed** [comment]
@bolyesz i think the latest RN version targets min api 21, lollipop. I think targeting API 23 will be fine for most people.

**Comment by bolyesz** [comment]
> @bolyesz i think the latest RN version targets min api 21, lollipop. I think targeting API 23 will be fine for most people.

thanks for your reply, I think you are right. Is there anything I can help with ?

**Comment by maurobender** [comment]
Hi! I'm having the same issue, this is being reported for an app that it's being deployed to Google Console and uses this library. Any way I can help getting new release with this fixed? Thanks!!

**Comment by rafetkhallaf** [comment]
@maurobender was your app rejected by Google for this?

**Comment by kylegillen** [comment]
Same. Anything we can do to help expedite a fix? Our app is still handily getting rejected and we’re blocked until we can resolve it.

Anyone know if this lib has the same issue: https://github.com/mrousavy/react-native-mmkv

**Comment by rafetkhallaf** [comment]
@nextriot Rejected you say... This is going to be a problem for everybody then. I have 3 apps that are already using this library, i find it easier to use than the other one, especially that it has the .map API.

I've used the other one before switching to this library. If you do end up testing it please let us know your findings as a lot of people will be in the same situation soon.

**Comment by ammarahm-ed** [comment]
I will fix this asap and release a new version.

**Comment by ammarahm-ed** [comment]
hey @rafetkhallaf @nextriot @bolyesz @maurobender I have fixed this on master. You can try to release on play store and let me know if the error is gone so i can release a new version of library.

```
npm install https://github.com/ammarahm-ed/react-native-mmkv-storage.git
```
or

```
yarn add npm install https://github.com/ammarahm-ed/react-native-mmkv-storage.git
```

**Comment by rafetkhallaf** [comment]
@ammarahm-ed great. I'll implement it and release an update today and get back to you.

**Comment by rafetkhallaf** [comment]
@ammarahm-ed Looks like it's good to go. No security error was reported in the pre-launch report details.

<img width="536" alt="Screenshot 2021-12-26 235247" src="https://user-images.githubusercontent.com/26126456/147448861-350bed7a-1301-468a-9b41-85db62e8e718.png">

**Comment by ammarahm-ed** [comment]
@rafetkhallaf Awesome I will release a new version to npm

**Comment by ammarahm-ed** [comment]
Released v0.6.7
